### PR TITLE
[Chatterbox] Bottlenecks optimization

### PR DESCRIPTION
**Pull request by Zbig9000**
**What problem does this PR solve?**
Chatterbox TTS autoregressive generation had several per-step C++ overheads that added up over a 1024-token loop: Ort::Session::Run was handed freshly-rebuilt const char* arrays, getInput/getOutput did O(N) linear searches, getInputNames/getOutputNames copied a ~60‑string vector by value on every call, selectNextToken allocated a vocab-sized buffer per token, the softmax used 4 passes over the vocab, FP16/Int4/UInt4 tensor reads materialised through a throwaway temp vector, and initInputTensors re-interrogated the ONNX graph (GetInputTypeInfo) for every input on every step.
Net effect on a 762-token English synthesis: ~3% of LM wall time was spent on setup/teardown around each ONNX run rather than on actual model work.

**How does it solve it?**
OnnxInferSession: cache inputNamesC_ / outputNamesC_ const char* arrays once in the constructor and pass them directly to Ort::Session::Run(); make getInput/getOutput O(1) via the existing inputIndexByName_ / outputIndexByName_ maps; change getInputNames/getOutputNames to return const std::vector<std::string>& (so binding-by-ref no longer copies); cache inputElementTypes_ in the constructor so initInputTensors skips the per-step GetInputTypeInfo FFI round-trip; reserve output and input tensor vectors.
IOnnxInferSession / OnnxInferSessionMock: updated interface to match the const-ref signature; tests switched from ::testing::Return(vec) to ::testing::ReturnRef(vec) / ReturnRefOfCopy as appropriate.
ChatterboxEngine: added reusable logitsBuffer_ / logitsBufferUncond_ members so selectNextToken and both CFG steps resize in place instead of allocating a ~26 KB vector per token; fused temperature scaling + softmax (applyTemperatureAndSoftmax) into 2 passes, and merged the min-P filter with its running sum (applyMinPFilterWithSum), skipping the redundant normalise pass since std::discrete_distribution renormalises internally; replaced readLastStepLogitsForBatch (returned a fresh vector) with readLastStepLogitsForBatchInto that writes into the caller's buffer; tidied assembleSpeechTokenSequence to reserve() once and drop the std::vector<int64_t>(3, SILENCE_TOKEN) temporary in favour of insert(end, 3, SILENCE_TOKEN); added <cmath>, <iterator>, <limits> headers for the fused numerics.
TensorFloatHandler: rewrote the Fp16, Int4, and UInt4 readToVector implementations so they grow the destination vector once and convert directly into its storage, eliminating the intermediate std::vector<float> converted(n) + copy.
Rejected during benchmarking: an earlier attempt to cache KV layout (lmKvInputNames_, kvChainedFlags_) at the engine level regressed LM generation by ~60%. Root cause: chainedInputNames_ in OnnxInferSession is only populated after the first Run() moves outputs into inputs, so caching chained flags right after enableKvCacheChaining() captured all-zeros and forced the hot loop to re-copy the full KV state every iteration. Reverted to the live isInputChained() lookup.

**How was it tested?**
C++ unit suite: 229/229 tests pass (9 skipped — pre-existing LavaSR / factory integration tests that need models not present locally). ChatterboxEngineMethodsTest, KvCacheChainingTest, Fp16UtilsTest, TensorFloatHandlerTest, OnnxInferSessionMockTest all cover the changed code paths.
End-to-end: CHATTERBOX_VARIANT=q4 bare examples/chatterbox-tts.js english produces a valid 5.4 s WAV (chatterbox-output.wav, 259244 bytes).
Benchmarked against main on a 762-token English synthesis (warm-up + 10 timed runs, q4 variant):
metric	before	after	Δ
LM generation
~10.79 s
~10.45 s
–3.1%
Total synthesize
~19.99 s
~19.65 s
–1.7%
The gain is entirely in C++ overhead around ORT (allocations, string hashing, FFI round-trips, redundant vector copies); ORT matmul time is untouched.

**API Changes**
IOnnxInferSession / OnnxInferSession metadata accessors are now return-by-const-reference. This is source-compatible for callers binding to a const reference or copying into a new vector, but binding to a non-const reference will no longer compile.

// Before
virtual std::vector<std::string> getInputNames() const = 0;
virtual std::vector<std::string> getOutputNames() const = 0;
// After
virtual const std::vector<std::string> &getInputNames() const = 0;
virtual const std::vector<std::string> &getOutputNames() const = 0;
Gmock users of OnnxInferSessionMock must switch from Return(vec) to ReturnRef(vec) (or ReturnRefOfCopy for temporaries) when stubbing these methods.

### CODE REVIEW

**Review by GustavoA1604 — CHANGES_REQUESTED** [review]
Update CHANGELOG.md as well
(could also include the changes of your last PR which I think didnt include them)
